libr/reg/arena.c

```diff
@@ -199,10 +199,11 @@ R_API int r_reg_arena_push(RReg *reg) {
 	int i;
 	for (i = 0; i < R_REG_TYPE_LAST; i++) {
 		RRegArena *a = reg->regset[i].arena;      // current arena
+		if (!a) continue;
 		RRegArena *b = r_reg_arena_new (a->size); // new arena
-		if (!a || !b) continue;
+		if (!b) continue;
 		// if (!i) {    r_print_hexdump (NULL, 0, a->bytes, a->size, 16, 16);    }
-		memcpy (b->bytes, a->bytes, a->size);
+		memcpy (b->bytes, a->bytes, b->size);
```

Comment:
okay. I guess behaviour of r_reg_arena_new might change and a->size will not be equal to b->size anymore